lxd/storage/backend_lxd_patches.go

```diff
@@ -124,3 +125,52 @@ func patchMissingSnapshotRecords(b *lxdBackend) error {
 
 	return nil
 }
+
+// patchCleanStorageVolumes deletes the remaining snapshot records in storage_volumes
+// (a previous patch would have already moved them into storage_volume_snapshots).
+func patchCleanStorageVolumes(b *lxdBackend) error {
+	var err error
+	var localNode string
+
+	err = b.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		localNode, err = tx.GetLocalNodeName(ctx)
+		if err != nil {
+			return fmt.Errorf("Failed to get local member name: %w", err)
+		}
+
+		return err
+	})
+	if err != nil {
+		return err
+	}
+
+	// Get instances on this local server (as the DB helper functions return volumes on local server), also
+	// avoids running the same queries on every cluster member for instances on shared storage.
+	filter := cluster.InstanceFilter{Node: &localNode}
+	err = b.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		err = b.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+			_, err = tx.Tx().Exec(`
+DELETE FROM storage_volumes AS sv WHERE sv.id IN (
+	SELECT sv.id FROM storage_volumes AS sv JOIN (
+		SELECT sv.name AS inst_name, svs.name AS snap_name, sv.project_id FROM storage_volumes AS sv
+		JOIN storage_volumes_snapshots AS svs
+			ON sv.id = svs.storage_volume_id
+	) j1 ON j1.project_id=sv.project_id
+	WHERE sv.name=FORMAT("%s/%s", j1.inst_name, j1.snap_name)
+);
+`)
+			if err != nil {
+				return fmt.Errorf("Failed to delete remaining instance snapshot records in the `storage_volumes` table: %w", err)
```

Comment:
> Failed deleting old instance storage volume snapshot records

Also this query isn't filtering by the instance storage volume type so that needs to be in there as well.